Source/Core/Core/PowerPC/JitCommon/JitBase.h

```diff
@@ -39,6 +41,48 @@
 #define JITDISABLE(setting)                                                                        \
   FALLBACK_IF(SConfig::GetInstance().bJITOff || SConfig::GetInstance().setting)
 
+struct RangeTracker
+{
+  rangeset<size_t> ranges;
+
+  void add(const u8* begin, const u8* end) { ranges.add((size_t)(begin), (size_t)(end)); }
+  void remove(const u8* begin, const u8* end) { ranges.remove((size_t)(begin), (size_t)(end)); }
+  void reset(const u8* begin, const u8* end)
+  {
+    ranges.clear();
+    add(begin, end);
+  }
+  std::optional<std::pair<u8*, u8*>> find_at_least(size_t size)
+  {
+    for (size_t i = 0; i < ranges.size(); ++i)
+    {
+      if (ranges.ivlen(i) >= size)
+      {
+        return std::make_pair((u8*)ranges.ivbegin(i), (u8*)ranges.ivend(i));
```

Comment:
These should be either explicit static_casts or reinterpret_casts since they're pointers.